### PR TITLE
fix: swap YT to UT logic

### PR DESCRIPTION
**Pull request by Senna46**
Fix SwapYTtoUT logic
Other functions are not changed.

### CODE REVIEW

File: x/irs/keeper/stripping.go [hunk 1]
```diff
@@ -113,11 +113,14 @@ func (k Keeper) RedeemPtYtPair(ctx sdk.Context, sender sdk.AccAddress, pool type
 	ptDenom := types.PtDenom(pool)
 	ytDenom := types.YtDenom(pool)
 
-	requiredPtAmount, err := k.CalculateRedeemRequiredPtAmount(ctx, pool, redeemUt)
+	requiredPtAmount, err := k.CalculateRedeemRequiredAmount(ctx, pool, redeemUt, ptDenom)
+	if err != nil {
+		return err
+	}
+	requiredYtAmount, err := k.CalculateRedeemRequiredAmount(ctx, pool, redeemUt, ytDenom)
```
Comment (Senna46) [inline x/irs/keeper/stripping.go:120]: @jununifi Restore the amount of YT required at the time of Redeem

```
PT = supply of PT * redeeming UT / total strategy supply
YT = supply of YT * redeeming UT / total strategy supply
```

File: x/irs/keeper/stripping.go [hunk 2]
```diff
@@ -148,10 +151,24 @@ func (k Keeper) CalculateRedeemRequiredPtAmount(ctx sdk.Context, pool types.Tran
 	if amountFromStrategy.IsZero() {
 		return sdk.ZeroInt(), types.ErrZeroAmount
 	}
-	ptDenom := types.PtDenom(pool)
-	ptSupply := k.bankKeeper.GetSupply(ctx, ptDenom)
-	requiredPtAmount := ptSupply.Amount.Mul(redeemUt).Quo(amountFromStrategy)
-	return requiredPtAmount, nil
+	supply := k.bankKeeper.GetSupply(ctx, denom)
+	requiredAmount := supply.Amount.Mul(redeemUt).Quo(amountFromStrategy)
```
Comment (jununifi) [inline x/irs/keeper/stripping.go:155]: Is it requriedYt amount or requiredUt amount? I think it would be good to clarify the token name on variable

Comment (Senna46) [reply x/irs/keeper/stripping.go:155]: requiredYtAmount or requiredPtAmount 
Supports both.

File: x/irs/keeper/stripping.go [hunk 3]
```diff
@@ -138,7 +141,7 @@ func (k Keeper) RedeemPtYtPair(ctx sdk.Context, sender sdk.AccAddress, pool type
 	return k.UnstakeFromStrategy(ctx, moduleAddr, sender.String(), pool.StrategyContract, redeemUt)
 }
 
-func (k Keeper) CalculateRedeemRequiredPtAmount(ctx sdk.Context, pool types.TranchePool, redeemUt sdk.Int) (sdk.Int, error) {
+func (k Keeper) CalculateRedeemRequiredAmount(ctx sdk.Context, pool types.TranchePool, redeemUt sdk.Int, denom string) (sdk.Int, error) {
```
Comment (jununifi) [inline x/irs/keeper/stripping.go:144]: Can you clarify the function name? RequiredYtAmount() or something like that

**Comment by Senna46** [comment]
Changed function name @jununifi

**Comment by Senna46** [comment]
Add query CLI support

**Pull request merged**